plugins/database/src/db_plugin.cpp

```diff
@@ -7089,64 +7196,223 @@ irods::error db_mod_user_op(
+                    }
+                    /* Group admin can only set the initial password, not update */
+                    return ERROR(CAT_INSUFFICIENT_PRIVILEGE_LEVEL, "insufficient privilege level");
+                }
+                rstrcpy(tSQL, form3, MAX_SQL_SIZE);
+                cllBindVars[cllBindVarCount++] = decoded;
+                cllBindVars[cllBindVarCount++] = myTime;
+                cllBindVars[cllBindVarCount++] = userIdStr;
+                if (logSQL != 0) {
+                    log_sql::debug("chlModUser SQL 9");
+                }
+            }
+            else {
+                opType = 4;
+                rstrcpy(tSQL, form4, MAX_SQL_SIZE);
+                cllBindVars[cllBindVarCount++] = userName2;
+                cllBindVars[cllBindVarCount++] = zoneName;
+                cllBindVars[cllBindVarCount++] = decoded;
+                cllBindVars[cllBindVarCount++] = "9999-12-31-23.59.01";
+                cllBindVars[cllBindVarCount++] = myTime;
+                cllBindVars[cllBindVarCount++] = myTime;
+                if (logSQL != 0) {
+                    log_sql::debug("chlModUser SQL 10");
+                }
+            }
+            status = cmlExecuteNoAnswerSql(tSQL, &icss);
         }
-        else {
-            opType = 4;
-            rstrcpy( tSQL, form4, MAX_SQL_SIZE );
-            cllBindVars[cllBindVarCount++] = userName2;
-            cllBindVars[cllBindVarCount++] = zoneName;
-            cllBindVars[cllBindVarCount++] = decoded;
-            cllBindVars[cllBindVarCount++] = "9999-12-31-23.59.01";
-            cllBindVars[cllBindVarCount++] = myTime;
-            cllBindVars[cllBindVarCount++] = myTime;
-            if ( logSQL != 0 ) {
-                log_sql::debug("chlModUser SQL 10");
+
+        // All the other operations use a common execution point for some reason. Setting a user's password is special
+        // because we may have to set values in two different database tables. As such, we execute the two statements
+        // above and commit once here, and then return early.
+        if (0 == status) {
+            status = cmlExecuteNoAnswerSql("commit", &icss);
+            if (status != 0) {
+                log_db::info("chlModUser cmlExecuteNoAnswerSql commit failure {}", status);
+                return ERROR(status, "commit failed");
             }
+            return SUCCESS();
         }
+
+        _rollback("chlModUser");
+
+        // This provides a more specific error code in the insertion case (as compared to the update case, which
+        // doesn't, for some reason).
+        if (4 == opType) {
+            std::vector<std::string> bindVars;
+            bindVars.emplace_back(userName2);
+            bindVars.emplace_back(zoneName);
+            const int ec = cmlGetIntegerValueFromSql(
+                "select user_id from R_USER_MAIN where user_name=? and zone_name=?", &iVal, bindVars, &icss);
+            if (0 != ec) {
```

Comment:
Probably want to check for `< 0` in case positive error codes make an appearance in the db plugin one day.

Reply:
Done